### DISCUSSION TITLE
Kana Input - missing 'he' (へ) and small "ya" (ゃ)

### DISCUSSION THREAD

**Discussion by takarabako-zz** [google/mozc#501] [Q&A]
I used to run Windows IME before, and I recently installed Mozc after switching to Ubuntu, and then I realized that in kana input, the kana へ is missing, but there are two kana of ほ (on English keyboard these are 'minus' key and 'equal' key).

Also the lowercase "ゃ" (from "や") is missing

And I could type more comfortably, if the middle row was: ちとしはきくまのりれけ, because I feel like I use the kana け more than the current "や", which is also available under the key with the number 7.

To visualize it, Mozc types out the kana like this (on my keyboard by Asus):
number row:  ゛ぬふあうえおやゆよわ**ほほ**
top row: たていすかんなにらせ゜むろ
middle row: ちとしはきくまのりれや
bottom row: つさそひこみもねるめ

and with SHIFT, it becomes:
number row:  をぬ゛ぁぅぇをぉけゅょろれ
top row: たてぃすかんなにらせ「」ー
middle row: ちとしはきくまのりけふ
bottom row: っさそひこみも、。・

_And this is what I wish the keyboard was like:_

without SHIFT:
number row: ？ぬふあうえおやゆよわほへ
top row: たていすかんなにらせをむろ
middle row: ちとしはきくまのりれけ
bottom row:つさそひこみもねるめ

with SHIFT:
number row: ！ぬふぁぅぇぉゃゅょ※ゐゑ
top row: たてぃすかんなにらせ「」ー
middle row: ちとしはきくまのり ゛ ゜
bottom row: っさそひこみも、。・

However, the proposed keyboard layout is only my wish, and nothing more. I wouldn't dare to tell you how to do your job, so I'll be more than glad if you could only fix the missing keys "ゃ" and "ほ".

Thank you for all your great work.
Have a Merry Christmas and a Happy New Year.

**Comment by takarabako-zz** [comment]
I forgot to include the version: It's Mozc (2.17.2116.102)

**Comment by shacharm2** [comment]
Ubuntu 18.04, ほほ issue still exists with shift.

**Comment by hiroyuki-komatsu** [comment]
For Kana input with US layout,

* If you use iBus, please modify ~/.config/mozc/ibus_config.textproto and change the `layout` from `default` to `us`. 
  (https://github.com/google/mozc/commit/9ba59b64d53365c1fe93c1c245b4ec3e35bdadf0)
* If you use Fcitx, please refer https://github.com/fcitx/mozc/issues/8.

**Reply by TomPlum** [reply]
Switching to the US layout fixes the issue, but the ` key is now ろ instead of voicing the last typed kana. The standard ゛and ゜keys still work, but I can't find how to type voiced kana glyphs with dakuten/handakutens

**Reply by hiroyuki-komatsu** [reply]
In the US layout, `゛` is mapped from `[` and `{`, and `゜` is mapped from `]` and `}`.

You can check the all mapping rules from
https://github.com/google/mozc/blob/master/src/unix/ibus/key_translator.cc#L264

**Reply by TomPlum** [reply]
I can't believe I never tried pressing `[` or `]` while I had a kana character active that had a diacritical form. Awesome, thanks for the explanation
